Foundation/NSFileManager.swift

```diff
@@ -712,9 +712,15 @@ extension FileManager {
 }
 
 extension FileManager {
-    open var homeDirectoryForCurrentUser: URL { NSUnimplemented() }
+    open var homeDirectoryForCurrentUser: URL {
+        return homeDirectory(forUser: CFCopyUserName().takeRetainedValue()._swiftObject)!
```

Comment:
@parkera  When you say asserting is the right answer do you mean we abort the application like this 
``` 
        let currentUser = CFCopyUserName().takeRetainedValue()._swiftObject
        guard !currentUser.isEmpty  else { fatalError ("$HOME not set") }
        return homeDirectory(forUser: currentUser)!  
```
I agree with @OscarSwanros . I cannot return nil as the declaration says that I cannot return nil.